W-13505366-aws-secrets-manager-kt (#2322)
mulesoft/docs-connectors: amazon-secrets-manager-properties-provider/1.0/modules/ROOT/pages/amazon-secrets-manager-properties-provider-reference.adoc.adoc

= Amazon Secrets Manager Properties Provider 1.0 Reference - Mule 4

> 

Reviewer: this file name has two .adocs in the name

> Azure Key Vault Properties Provider is a configuration that enables you to retrieve your secrets from Amazon Secrets Manager cloud service before your Mule app runs.

Reviewer: ```suggestion
Amazon Secrets Manager Properties Provider is a configuration that enables you to retrieve your secrets from Amazon Secrets Manager cloud service before your Mule app runs.
```

[[Config]]
== Configuration

Default Configuration for Amazon Secrets Manager Properties Provider.

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Name | Type | Description | Default Value | Required
|Name | String | Name for this configuration. Connectors reference the configuration with this name. | | x
| Expiration Policy a| <<ExpirationPolicy>> |  Configures the minimum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. This does not mean that the instance expires at the exact moment that it becomes eligible. Mule purges the instances as appropriate. |  | x

> | AWS Secrets Manager Region a| String |  AWS Secrets Manager region, as us-east-2. |  | x

Reviewer: ```suggestion
| AWS Secrets Manager Region a| String |  AWS Secrets Manager region, for example, `us-east-2`. |  | x
```

| AWS Access Key a| String |  AWS access key ID. |  | 
| AWS Secret Key a | String |  AWS secret key ID. |  | 

> |Role ARN a| String | The Role ARN unique identifies a role to gain cross account access. |  |

Reviewer: ```suggestion
|Role ARN a| String | Uniquely identifies a role to gain cross-account access. |  |
```

|===

== Types
[[ExpirationPolicy]]
=== Expiration Policy

[cols=".^20%,.^25%,.^30%,.^15%,.^10%", options="header"]
|======================
| Field | Type | Description | Default Value | Required
| Max Idle Time a| Number | Configures the maximum amount of time that a dynamic configuration instance can remain idle before Mule considers it eligible for expiration. |  | 
| Time Unit a| Enumeration, one of:

** NANOSECONDS
** MICROSECONDS
** MILLISECONDS
** SECONDS
** MINUTES
** HOURS

> ** DAYS | A time unit that qualifies the maxIdleTime attribute |  | 

Reviewer: ```suggestion
** DAYS | Time unit for the *Max Idle Time* field. |  | 
```
